lib/msf/core/opt_enum.rb

```diff
@@ -25,7 +25,7 @@ def valid?(value = self.value, check_empty: true)
     end
 
     def normalize(value = self.value)
-      if valid?(value)
+      if valid?(value) && !value.nil?
```

Comment:
Does this need to honor the semantics of `empty_required_value?` and `required?` similar to the current `valid?` implementation?

```
    def valid?(value = self.value, check_empty: true)
      return false if check_empty && empty_required_value?(value)
      return true if value.nil? && !required?
      !value.nil? && enums.include?(value.to_s)
    end
```

Reply:
It seems fine as-is to me, I added some tests to make sure things are working right, but let me know if you think I've got some of the logic backwards